data/trie/patriciaMerkleTrie_test.go

```diff
@@ -327,6 +327,21 @@ func emptyTrie() data.Trie {
 	return tr
 }
 
+func TestPatriciaMerkleTrie_DeepCloneShouldWork(t *testing.T) {
+	tr := initTrie()
+
+	_ = tr.Update([]byte("doee"), []byte("value of doee"))
+	_ = tr.Update([]byte("doeee"), []byte("value of doeee"))
+
+	trie2, err := tr.DeepClone()
+
+	assert.Nil(t, err)
+
+	assert.Equal(t, tr, trie2)
+	assert.False(t, tr == trie2)
+	assert.Equal(t, tr.String(), trie2.String())
+}
```

Comment:
As another method to see if the clone was done correctly, you could compute the rootHash of both tries, and then compare the hashes.

Reply:
I think `assert.Equal(t, tr, trie2)` will suffice. However, I have added the assert as a supplementary check against RootHash algorithm